controllers/users.js

```diff
@@ -3,25 +3,31 @@ const { pixelateImageFromURL } = require("../middleware/imagePixelation");
 const { ROUTES } = require("./routes");
 
 const { cardStyle } = require("../public/javascripts/extraStyles.js");
+const { dataSchemaValidation } = require("../middleware");
+const { registrationSchema } = require("../schemas");
 
 module.exports.renderRegister = async (req, res) => {
   const image = await pixelateImageFromURL(req.gameData.background_image);
   res.render("users/register.ejs", { image, extraStyles: cardStyle });
 };
 
 module.exports.register = async (req, res, next) => {
-  try {
-    const { email, username, password } = req.body;
-    const user = new User({ email, username });
-    const registeredUser = await User.register(user, password);
-    req.login(registeredUser, (err) => {
-      if (err) return next(err);
-      req.flash("success", "Welcome!");
-      res.redirect(ROUTES.index);
-    });
-  } catch (e) {
-    req.flash("error", e.message);
-    res.redirect(ROUTES.register);
+  if (dataSchemaValidation(req.body, registrationSchema)) {
```

Comment:
Ok so validator works I assume, next step would probably be to introduce it as a middleware as having if checks inside every functions body imo a bit redundant. I found a resource where you could look at some examples how to implement it: https://softchris.github.io/pages/joi.html#building-a-middleware